File: designer/types/condition.ts
export type XComparison = "IS" | "NOT";
export type XOperator = "ALL" | "ANY";

export type XCondition = IConditionSingle | IConditionMany

export interface IConditionArea {
	area: string;
	condition: XCondition;
}

export interface IConditionItem {
	item: string;
	condition: XCondition;
}

export interface IConditionSingle {
	left: string;
	comparison: XComparison;
	right: string;
	operator?: never;
	conditions?: never;
};

export interface IConditionMany {
	left?: never;
	comparison?: never;
	right?: never;
	operator: XOperator;
	conditions: XCondition[];
};

File: api/BoningerWorks.TextAdventure.Json/Inputs/Condition.cs
using BoningerWorks.TextAdventure.Json.Utilities;
using System.Text.Json.Serialization;

namespace BoningerWorks.TextAdventure.Json.Inputs
{
	public class Condition
	{
		[JsonPropertyName("left")] public string? Left { get; set; }
		[JsonPropertyName("comparison")] public string? Comparison { get; set; }
		[JsonPropertyName("right")] public string? Right { get; set; }
		[JsonPropertyName("operator")] public string? Operator { get; set; }
		[JsonPropertyName("conditions")] public OneOrManyList<Condition?>? Conditions { get; set; }
	}
}
